### ISSUE TITLE
Custom k3s_binary_url

### ISSUE THREAD

**Issue by bconn98** [PyratLabs/ansible-role-k3s#211]
### Summary

Allow for overriding of `k3s_binary_url` for use in FOSS controlled environments. My employer requires that we pull FOSS components from a company controlled nexus mirror and this does not appear to be supported by the role currently.

### Issue Type

- Feature Request

### User Story

_As an_ ansible user, _I want to_ override the `k3s_binary_url` variable _so that_ I can pull from sources other than GitHub.

**Comment by onedr0p** [comment]
We're open to PRs on that implementation.

**Comment by xanmanning** [comment]
> We're open to PRs on that implementation.

agree, happy to take a look if you have a solution in mind to implement

**Comment by jon-stumpf** [comment]
I have this implemented on my fork.  Since @dereknola has become a maintainer here, I have been working to get my commits merged in the main branch.  My latest working branch can be found [here](https://github.com/jon-stumpf/k3s-ansible/tree/rebase-upstream).  I have been busy with other things for the last few weeks but this is a stable branch.  Let me know if this works for you or if you need any changes.

**Comment by dbrennand** [comment]
Implemented logic for this (untested) in https://github.com/dbrennand/ansible-role-k3s/commit/56db9e894eeeee5182b350150e10a85e3a5179c0.

Usage:

```
k3s_airgap: true
k3s_binary_url: https://nexus.company.com/path/to/k3s
k3s_hash_url: https://nexus.company.com/path/to/checksum.txt
```

**Comment by onedr0p** [comment]
That implementation seems fine to me.
